### PR TITLE
Fix package resolver score

### PR DESCRIPTION
**Pull request by carlobaldassi**
This fixes the first part of #20313, by allowing for packages to be installed without penalizing other packages. In other words: previously, installing a package could have an unintended "cost" under some particular circumstances. This could have happened when the solver had to choose between installing an earlier version of a package "A" by itself, or a newer version of "A" with a dependency "B", and when the candidate version for "B" was not the highest possible. The first commit in this PR fixes that. The other commits are a bit of housekeeping.
cc @tkelman

### CODE REVIEW

File: base/pkg/resolve/fieldvalue.jl [hunk 1]
```diff
@@ -55,13 +55,15 @@ function Base.isless(a::FieldValue, b::FieldValue)
...
     return false
 end
 
-==(a::FieldValue, b::FieldValue) =
-    a.l0 == b.l0 && a.l1 == b.l1 && a.l2 == b.l2 && a.l3 == b.l3 && a.l4 == b.l4
+Base.:(==)(a::FieldValue, b::FieldValue) =
+    a.l0 == b.l0 && a.l1 == b.l1 && a.l2 == b.l2 && a.l3 == b.l3 && a.l4 == b.l4 && a.l5 == b.l5
 
-Base.abs(a::FieldValue) = FieldValue(abs(a.l0), abs(a.l1), abs(a.l2), abs(a.l3), abs(a.l4))
+    Base.abs(a::FieldValue) = FieldValue(abs(a.l0), abs(a.l1), abs(a.l2), abs(a.l3), abs(a.l4), abs(a.l5))
```
Comment (tkelman) [inline base/pkg/resolve/fieldvalue.jl]: shouldn't be indented?

**Review by tkelman — APPROVED** [review]
lgtm aside from trivial indentation in one place

do you want to try this on master or wait a bit and see if there are other improvements you can make to the reporting?

File: base/pkg/resolve/fieldvalue.jl (hunk 1 above)
Comment (carlobaldassi) [inline base/pkg/resolve/fieldvalue.jl]: oops, fixed, thanks.

**Comment by carlobaldassi** [comment]
I think that any improvements in the reporting would be orthogonal to this PR, so this can be merged in the meanwhile.

**Pull request merged**